include/pybind11/pytypes.h

```diff
@@ -774,6 +782,8 @@ class list : public object {
     size_t size() const { return (size_t) PyList_Size(m_ptr); }
     detail::list_accessor operator[](size_t index) const { return {*this, index}; }
     void append(handle h) const { PyList_Append(m_ptr, h.ptr()); }
+    template <typename T, detail::enable_if_t<detail::is_implicitly_castable<T>::value, int> = 0>
+    void append(const T &value) const;
```

Comment:
I wonder if this and `operator=` can be streamlined a bit so that they don't need both the `handle` and template `T` overloads? Having just the template overload without the extra SFINAE would be ideal IMO.

I guess the main issue here is that a `py::cast` on `handle` would introduce an unnecessary `incref/decref`, so perhaps there should be something more specialized here. This could be useful for future expansion of the `py::object`-derived classes. There are a lot of methods which could be exposed in C++ similarly to `list.append()` and it would be unfortunate to have to create 2 overloads for each new method.